File: src/main/java/co/edu/javeriana/dw/proyecto/model/Spacecraft.java
package co.edu.javeriana.dw.proyecto.model;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@EqualsAndHashCode(of = "id")
public class Spacecraft {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String name;
    private BigDecimal credit;
    private Double totalTime;

    @OneToMany(mappedBy = "spacecraft")
    private List<Player> players;

    @ManyToOne
    @JoinColumn(name = "planet_id")
    private Planet planet;

    @OneToOne
    private SpacecraftModel spacecraftModel;

    @OneToMany(mappedBy = "spacecraft")
    private List<Inventory> inventories;
}

File: Proyecto-Desarrollo-Web-Client/src/app/model/spacecraft.ts
import { Inventory } from "./inventory";
import { Planet } from "./planet";
import { Player } from "./player";
import { SpacecraftModel } from "./spacecraft-model";

export class Spacecraft {
  constructor(
  public id: number,
  public name: string, 
  public credit: number,
  public totalTime: number,
  public players : Player[],
  public planet: Planet,
  public spacecraftModel: SpacecraftModel,
  public inventories: Inventory[]
  ){}
}
